docs/guide/02-Introducing-Components.md

```diff
@@ -0,0 +1,316 @@
+
+If your component takes input, then you should describe it with a type. For example, a component that accepts an integer as input would use this type:
+
+```purs
+type Input = Int
+```
+
+Our counter doesn't require any input, so we have two choices. First, we can just say that our input type is `Unit`, meaning that we'll just take a dummy value and throw it away:
+
+```purs
+type Input = Unit
+```
+
+Second, and more commonly, anywhere our input type shows up in our component we can simply leave it as a type variable: `forall i. ...`. It's perfectly fine to use either approach, but from here on out we'll use type variables to represent types our component isn't using.
+
+### State
+
+Halogen components maintain an internal state over time, which is used to drive the component's behavior and to produce HTML. Our counter component maintains the current count, an integer, so we'll use that as our state type:
+
+```purs
+type State = Int
+```
+
+Our component needs to also produce an initial state value. All Halogen components require an `initialState` function which produces the initial state from the input value:
+
+```purs
+initialState :: Input -> State
+```
+
+Our counter component doesn't use its input, so our `initialState` function won't use an input type and will instead just leave that type variable open. Our counter should start at 0 when the component runs.
+
+```purs
+initialState :: forall i. i -> State
+initialState _ = 0
+```
+
+### Actions
+
+Halogen components can update state, perform effects, and communicate with other components in response to events that arise internally. Components use an "action" type to describe what kinds of things a component can do in response to internal events.
+
+Our counter has two internal events: a click event on the button to decrement the count and a click event on the button to increment the count. We can describe what our component should do in response to these events using a data type we'll call `Action`:
+
+```purs
+data Action = Increment | Decrement
+```
+
+This type signifies that our component is capable of incrementing and decrementing. In a moment, we'll see this type used in our HTML -- another example of Halogen's declarative nature.
+
+Just like how our state type had to be paired with an `initialState` function that describes how to produce a `State` value, our `Action` type should be paired with a function called `handleAction` that describes what to do when one of these actions occurs.
+
+```purs
+handleAction :: forall o m. Action -> H.HalogenM State Action () o m Unit
+```
+
+As with our input type, we can leave type variables open for types that we aren't using. 
+
+* The type `()` means our component has no child components. We could also leave it open as a type variable, because we aren't using it, but `()` is so short that you'll see this commonly used.
```

Comment:
```suggestion
* The type `()` means our component has no child components. We could also leave it open as a type variable because we aren't using it -- `slots`, by convention -- but `()` is so short you'll see this type commonly used instead.
```